File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/mapper/IngredienteMapper.java
package mapper;

import dto.ingrediente.IngredienteDto;
import dto.ingrediente.ListaIngredienteDto;
import models.Ingrediente;

import java.util.List;

public class IngredienteMapper {

    public static IngredienteDto toDto(Ingrediente ingrediente){
        return new IngredienteDto(
                String.valueOf(ingrediente.getId()),
                ingrediente.getNombre(),
                String.valueOf(ingrediente.getPrecio())
        );
    }

    public static Ingrediente toIngrediente(IngredienteDto ingredienteDto){
        return new Ingrediente(
                Integer.parseInt(ingredienteDto.getId()),
                ingredienteDto.getNombre(),
                Double.parseDouble(ingredienteDto.getPrecio())
        );
    }

    public static ListaIngredienteDto toDto(List<Ingrediente> ingredientes){
        return new ListaIngredienteDto(ingredientes.stream().map(ingrediente -> toDto(ingrediente)).toList());
    }

    public static List<Ingrediente> toIngredientes(ListaIngredienteDto ingredientes){
        return ingredientes.ingredientesDto.stream().map(i -> toIngrediente(i)).toList();
    }

}

File: buerguer-pig/JCZhang-BurguerPigFicheros-Kotlin/src/main/kotlin/mappers/ingredienteMapper.kt
package mappers

import dto.IngredienteDTO
import dto.IngredientesListDto
import models.Ingrediente



fun IngredienteDTO.toIngrediente() = Ingrediente(
    ID = this.id,
    nombre = this.nombre,
    precio = this.precio
)

fun Ingrediente.toDto() = IngredienteDTO(
    id = ID,
    nombre = nombre,
    precio = precio

)

fun List<Ingrediente>.ingredienteToDto() = IngredientesListDto(
    ingredientes = map{it.toDto()}
)

fun IngredientesListDto.toIngredienteList() = ingredientes.map { it.toIngrediente() }